exonum/src/blockchain/config.rs

```diff
@@ -87,16 +87,30 @@ impl ConsensusConfig {
     /// Default value for max_message_len.
     pub const DEFAULT_MAX_MESSAGE_LEN: u32 = 1024 * 1024; // 1 MB
 
-    /// Checks if propose timeout is less than round timeout. Warns if fails.
+    /// Produces warnings if configuration contains non-optimal values.
+    ///
+    /// Validation for logical correctness is performed in the `StoredConfiguration::try_deserialize`
+    /// method, but some values can decrease consensus performance.
     #[doc(hidden)]
-    pub fn validate_configuration(&self) {
+    pub fn warn_nonoptimal(&self) {
```

Comment:
IMHO `warn_nonoptimal` doesn't reflect the essence of this method. It doesn't simply warn, it warns only if something is wrong. Maybe something like `warn_if_nonoptimal`?